### PR TITLE
Adding utilities to score mondo grounding predictions

### LINKED ISSUE #14: Add mondo scoring utilities
TODO: Add picture of whiteboard from hackathon

For example, given this structure:

image

Where the highlighted mondo is an OMIM PS, and the leaf nodes correspond to OMIM:

|subject_id|predicate_id|object_id|subject_label|
|---|---|---|---|
|MONDO:0024530|oio:hasDbXref|OMIM:158810|Bethlem myopathy 1A|
|MONDO:0024530|skos:exactMatch|OMIM:158810|Bethlem myopathy 1A|
|MONDO:0034022|oio:hasDbXref|OMIM:616471|Bethlem myopathy 2|
|MONDO:0034022|skos:exactMatch|OMIM:616471|Bethlem myopathy 2|
|MONDO:0958233|oio:hasDbXref|OMIM:620725|Bethlem myopathy 1B|
|MONDO:0958233|skos:exactMatch|OMIM:620725|Bethlem myopathy 1B|
|MONDO:0958234|oio:hasDbXref|OMIM:620726|Bethlem myopathy 1C|
|MONDO:0958234|skos:exactMatch|OMIM:620726|Bethlem myopathy 1C|

(ignore the duplicative xrefs)

We expect

```python
    >>> score_grounded_result("MONDO:0008029", "OMIM:158810")
    0.5
```

This example is Bethlem myopathy (MONDO:0008029) which groups
BTHLM1A (OMIM:158810) and others. We score this 0.5

### PR DESCRIPTION
**Pull request by cmungall**
Fixes #14

### CODE REVIEW

File: src/malco/post_process/mondo_score_utils.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        term (str): The term.
+
+    Returns:
+        str: The OMIM mappings.
+    """
+    adapter = mondo_adapter()
+    if not isinstance(adapter, MappingProviderInterface):
+        raise ValueError("Adapter is not an MappingProviderInterface")
+    omims = []
+    for m in adapter.sssom_mappings([term], "OMIM"):
```
Comment (joeflack4) [inline src/malco/post_process/mondo_score_utils.py:43]: Do we only need `OMIM` here and not also `OMIMPS`?

**Comment by cmungall** [comment]
Do any phenopackets use the PS as a diagnosis? I didn’t think so

**Comment by justaddcoffee** [comment]
FWIW these are the 209 unique disease IDs mentioned in `diagnosis` of any phenopacket in the latest release of phenopacket-store:
https://www.file.io/jOJA/download/Ym1q0sywZf32
no `OMIMPS` IDs

**Comment by cmungall** [comment]
https://www.file.io/error

**Pull request merged**